ios/ChunkManager.mm

```diff
@@ -29,87 +29,93 @@ @implementation ChunkManager
-    }
+        // Handle http & https
+        if ([[chunkUrl scheme] hasPrefix:@"http"]) {
+            if (fetch) {
+                [self downloadAndCache:chunkId chunkUrl:chunkUrl completionHandler:^(NSError *error) {
+                    if (error) {
+                        reject(ChunkDownloadFailure, error.localizedFailureReason, nil);
+                    } else {
+                        [self execute:bridge chunkId:chunkId url:chunkUrl withResolver:resolve withRejecter:reject];
+                    }
+                }];
+            } else {
+                [self execute:bridge chunkId:chunkId url:chunkUrl withResolver:resolve withRejecter:reject];
+            }
+            
+        } else if ([[chunkUrl scheme] isEqualToString:@"file"]) {
+            [self executeFromFilesystem:bridge url:chunkUrl withResolver:resolve withRejecter:reject];
+            
+        } else {
+            reject(UnsupportedScheme,
+                   [NSString stringWithFormat:@"Scheme in URL '%@' is not supported", chunkUrlString], nil);
+        }
+    }];
 }
 
 RCT_EXPORT_METHOD(preloadChunk:(nonnull NSString*)chunkId
                   chunkUrl:(nonnull NSString*)chunkUrlString
                   fetch:(BOOL)fetch
                   withResolver:(RCTPromiseResolveBlock)resolve
-                  withRejecter:(RCTPromiseRejectBlock)reject) {
-    if (!fetch) {
-        // Do nothing, chunk is already preloaded
-        resolve(nil);
-    } else {
-        NSURL *chunkUrl = [NSURL URLWithString:chunkUrlString];
-        if ([[chunkUrl scheme] hasPrefix:@"http"]) {
-            [self downloadAndCache:chunkId chunkUrl:chunkUrl completionHandler:^(NSError *error) {
-                if (error) {
-                    reject(ChunkDownloadFailure, error.localizedFailureReason, nil);
-                } else {
-                    resolve(nil);
-                }
-            }];
+                  withRejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self runInBackground:^(){
+        if (!fetch) {
+            // Do nothing, chunk is already preloaded
+            resolve(nil);
         } else {
-            reject(UnsupportedScheme,
-                   [NSString stringWithFormat:@"Scheme in URL '%@' is not supported", chunkUrlString], nil);
+            NSURL *chunkUrl = [NSURL URLWithString:chunkUrlString];
+            if ([[chunkUrl scheme] hasPrefix:@"http"]) {
+                [self downloadAndCache:chunkId chunkUrl:chunkUrl completionHandler:^(NSError *error) {
+                    if (error) {
+                        reject(ChunkDownloadFailure, error.localizedFailureReason, nil);
+                    } else {
+                        resolve(nil);
+                    }
+                }];
+            } else {
+                reject(UnsupportedScheme,
+                       [NSString stringWithFormat:@"Scheme in URL '%@' is not supported", chunkUrlString], nil);
+            }
```

Comment:
I sugget to change it to:
```objc
if (!fetch) {
  // Do nothing, chunk is already preloaded
  resolve(nil);
} else {
  [self runInBackground:^() {
    ...
  }];
}
```
To avoid unnecessarily spawning new threads.

Reply:
Done.